pkg/runtime/grpc/oci.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcruntime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+func (r *Runtime) GetOCIGadgetInfo(gadgetCtx runtime.GadgetContext) (*api.GadgetInfo, error) {
+	timeout := time.Second * time.Duration(r.globalParams.Get(ParamConnectionTimeout).AsUint16())
+	ctx, cancelDial := context.WithTimeout(gadgetCtx.Context(), timeout)
+	defer cancelDial()
+
+	// use default params for now
+	params := r.ParamDescs().ToParams()
+	conn, err := r.getConnToRandomTarget(ctx, params)
+	if err != nil {
+		return nil, fmt.Errorf("dialing random target: %w", err)
+	}
+	defer conn.Close()
+	client := api.NewOCIGadgetManagerClient(conn)
+
+	paramMap := make(map[string]string)
+
+	// OCI exists both on client and server
+	gadgetCtx.LocalOperatorsParamCollection()["oci"].CopyToMap(paramMap, "oci.")
```

Comment:
Why oci params are called "local"?

Reply:
A distinction is made between locally running operators (like the CLIOperator) and remote operators (like KubeManager etc.). For now, the OCI handler also registers some local params, but sends those over as well. I see this (and the hard reference to "oci") as a hack for now that can be improved later on (maybe give it a dedicated place inside the gadgetContext, for example - not 100% sure).

Btw: an update to param handling will land soon, I'm still working on that.

Reply:
Mmmm the OCI handler is a remote "operator" and it registers some params. Those params are being stored in gadgetContext like "localoperatorparams". That word "local" confused me at the beginning because I was expecting the OCI handler to run also locally, but it's not.

I suppose what you're trying to distinguish is that the OCI handler params are pulled "statically", while other remote operator params, like `--containername` and `--namespace` for KubeManager are pulled with the `GetOCIGadgetInfo`. Am I right?

I'm sorry, I'm just trying to be sure I understood things properly.